### PR TITLE
Bugsnag undefined coords error

### LINKED ISSUE #394: undefined is not an object (evaluating 'Be.coords') in app/components/Common/MapMarking/index.js:284
## Error in Tree Mapper

**TypeError** in **MainActivity**
undefined is not an object (evaluating 'Be.coords')

[View on Bugsnag](https://app.bugsnag.com/plant-for-the-planet/tree-mapper/errors/60ddc0f0bdb7710007f3ac14?event_id=60ddc0f0007f9ad430570000&i=gh&m=ci)

## Stacktrace

    app/components/Common/MapMarking/index.js:284 - 
    app/components/Common/MapMarking/index.js:283 - 

[View full stacktrace](https://app.bugsnag.com/plant-for-the-planet/tree-mapper/errors/60ddc0f0bdb7710007f3ac14?event_id=60ddc0f0007f9ad430570000&i=gh&m=ci)

*Created automatically via Bugsnag*

### PR DESCRIPTION
**Pull request by tejassonar**
Fixes #394, #391, #358, #346, #345, #328

Changes in this pull request:
- Fixed location.coords

@norbertschuler

### CODE REVIEW

**Comment by norbertschuler** [comment]
@tejassonar do you have an idea how to reproduce this error. I am not sure when it happens and what exactly went wrong?

**Comment by tejassonar** [comment]
> @tejassonar do you have an idea how to reproduce this error. I am not sure when it happens and what exactly went wrong?

Not sure. But, when I saw the error on bugsnag, I found out that it might be due to the state management of React. The location took a little longer to set, which is why it was showing 'undefined'. So instead of using location from the state, I resolved position from the function 'updateCurrentPosition' and used it.

File: app/components/Common/MapMarking/index.js [hunk 1]
```diff
@@ -415,7 +415,7 @@ export default function MapMarking({
           setAccuracyInMeters(position.coords.accuracy);
           onUpdateUserLocation(position);
           setLocation(position);
```
Comment (norbertschuler) [inline app/components/Common/MapMarking/index.js:417]: Do you think setting the retrieved location in the state variable is happening async and thus location can still be undefined above? Thus returning it as a local variable makes the information available at once?

**Review by norbertschuler — APPROVED** [review]
If I understood the issue correctly (see above), this fix should resolve it.

File: app/components/Common/MapMarking/index.js (hunk 1 above)
Comment (norbertschuler) [reply app/components/Common/MapMarking/index.js:417]: Answered already above ;-)

Comment (tejassonar) [reply app/components/Common/MapMarking/index.js:417]: Yes. You framed it in a sophisticated way.

**Pull request merged**